### DISCUSSION TITLE
Typescript error after upgrade from 6.1.12 to 6.2.2

### DISCUSSION THREAD

**Discussion by Valeronlol** [mikro-orm/mikro-orm#5493] [General]
### Describe the bug

We just upgraded version from 6.1.12 -> 6.2.2 and got TS error "Type instantiation is excessively deep and possibly infinite."
Downgrade resolves this issue.

### Reproduction

pretty basic case:
`await this.em.populate(entity, ['relation1', 'relation2'])`

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.2.2

### Node.js version

20.12

### Operating system

alpine

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
Please provide a complete reproduction, could be caused by TS config or other things. One line without any context is not a reproduction. If it were so simple to reproduce, it would be surfacing in the existing ORM test suite (which is now almost 3000 tests, many of them doing something like what you say breaks), and they all compile just fine.
